### PR TITLE
luminous: ceph-kvstore-tool: use unique_ptr<> manage the lifecycle of bluestore

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
the cli test failure is expected. i don't have plan to fix it.

File: src/tools/ceph_kvstore_tool.cc [hunk 1]
```diff
@@ -39,8 +39,9 @@ class StoreTool
 #ifdef HAVE_LIBAIO
   struct Deleter {
     BlueStore *bluestore;
-    Deleter(BlueStore *store = nullptr)
-    : bluestore(store)
+    Deleter() {}
```
Comment (ifed01) [inline src/tools/ceph_kvstore_tool.cc:42]: I might want to assign bluestore with nullptr

Comment (tchaikov) [reply src/tools/ceph_kvstore_tool.cc]: @ifed01 fixed and repushed.

**Review by ifed01 — APPROVED** [review]

**Pull request merged**